src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +138,59 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
```

Comment:
We're purposely not specifying comparer here because ordinal is wrong for this case, and invariant doesn't exist on .NET Core. The only options on .NET Core are CurrentCulture and Ordinal/OrdinalIgnoreCase.